modules/kernel/src/main/scala-2.10/notebook/ReplCalculator.scala

```diff
@@ -390,15 +391,16 @@ class ReplCalculator(
         val cellId = er.cellId
         def replEvaluate(code:String, cellId:String) = {
           val cellResult = try {
-           repl.evaluate(s"""
+            repl.evaluate(s"""
               |sparkContext.setJobGroup("${JobTracking.jobGroupId(cellId)}", "${JobTracking.jobDescription(code)}")
               |$code
               """.stripMargin,
-              msg => thisSender ! StreamResponse(msg, "stdout")
+              msg => thisSender ! StreamResponse(msg, "stdout"),
+              (termOrType, tpe, references) => thisSender ! DefinitionResponse(termOrType, tpe, cellId, references)
```

Comment:
nicez. so to know if last code line is a value definition or a "return" value, we simply check if cell has defined any `resXY` variable. will do some good for #434